extensions/nodes/node_status.go

```diff
@@ -132,6 +133,34 @@ func AllNodeDeleted(client *rancher.Client, ClusterID string) error {
 
 			if len(nodes.Data) == 0 {
 				logrus.Infof("All nodes in the cluster are deleted!")
+
+				return true, nil
+			}
+
+			return false, nil
+		})
+
+	return err
+}
+
+// IsNodeDeleted is a helper method that will loop and check if the node is deleted in the cluster.
+func IsNodeDeleted(client *rancher.Client, nodeName, ClusterID string) error {
+	ctx := context.Background()
+	err := wait.PollUntilContextTimeout(
+		ctx, oneSecondInterval, defaults.ThirtyMinuteTimeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
noting again 30 minutes seems long and could use a comment explaining why.

Reply:
Resolved